### PR TITLE
GEODE-1648: Introduce security-enabled-components and tests

### PR DESCRIPTION
**Pull request by gracemeilen**

### CODE REVIEW

File: geode-core/src/main/java/com/gemstone/gemfire/distributed/internal/AbstractDistributionConfig.java [hunk 1]
```diff
@@ -498,6 +502,29 @@ protected String checkRedisBindAddress(String value) {
     return value;
   }
 
+  /**
+   * First check if sslComponents are in the list of valid components. If so, check that no other *-ssl-* properties other than cluster-ssl-* are set.
```
Comment (jinmeiliao) [inline geode-core/src/main/java/com/gemstone/gemfire/distributed/internal/AbstractDistributionConfig.java]: comments need to be changed here.

File: geode-core/src/main/java/com/gemstone/gemfire/internal/security/GeodeSecurityUtil.java [hunk 2]
```diff
@@ -471,16 +499,27 @@ public static PostProcessor getPostProcessor() {
     return postProcessor;
   }
 
-  public static boolean isClientSecurityRequired() {
-    return isClientAuthenticator || isIntegratedSecurity;
+  public static boolean isIntegratedSecurity(){
```
Comment (jinmeiliao) [inline geode-core/src/main/java/com/gemstone/gemfire/internal/security/GeodeSecurityUtil.java]: We can get rid of this public method now. I believe now all the other components are just using component specific security enabled function.

File: geode-core/src/main/java/com/gemstone/gemfire/management/internal/ManagementAgent.java [hunk 3]
```diff
@@ -447,8 +449,7 @@ public synchronized void start() throws IOException {
       }
     };
 
-    String shiroConfig = this.config.getShiroInit();
-    if (! StringUtils.isBlank(shiroConfig) || isIntegratedSecurity()) {
+    if (isIntegratedSecurity()) {
```
Comment (jinmeiliao) [inline geode-core/src/main/java/com/gemstone/gemfire/management/internal/ManagementAgent.java]: probably be easier to read if it just read "if (GeodeSecurityUtil.isJmxSecurityEnabled())" here.

File: geode-core/src/main/java/com/gemstone/gemfire/management/internal/ManagementAgent.java [hunk 4]
```diff
@@ -509,8 +510,7 @@ private void registerAccessControlMBean() {
 
 
   private boolean isIntegratedSecurity() {
```
Comment (jinmeiliao) [inline geode-core/src/main/java/com/gemstone/gemfire/management/internal/ManagementAgent.java]: We can get rid of this method. I believe this is called to check if we need to enabled pulse security in the same file as well. Yeah, what about pulse security? Do we need to add another enum or just say if jmx is enabled. pulse should be enabled as well?

**Comment by kirklund** [comment]
1) we need to fix the javadocs for security-enabled-components on DistributedSystem class
2) move SecurableComponent enum into internal.security pkg
3) consider collapsing SecurableComponents interface into SecurableComponent enum (we don't really need both

**Pull request merged**